src/category/equiv_functor.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Scott Morrison
+-/
+
+import category_theory.category
+import data.equiv.functor
+
+/-!
+# Functions functorial with respect to equivalences
+
+An `equiv_functor` is a function from `Type → Type` equipped with the additional data of
+coherently mapping equivalences to equivalences.
+
+In categorical language, it is an endofunctor of the "core" of the category `Type`.
+-/
+
+universes u₀ u₁ u₂ v₀ v₁ v₂
+
+open function
+
+/--
+An `equiv_functor` is only functorial with respect to equivalences.
+
+To construct an `equiv_functor`, it suffices to supply just the function `f α → f β` from
+an equivalence `α ≃ β`, and then prove the functor laws. It's then a consequence that
+this function is part of an equivalence, provided by `equiv_functor.map_equiv`.
+-/
+class equiv_functor (f : Type u₀ → Type u₁) :=
+(map : Π {α β}, (α ≃ β) → (f α → f β))
+(id_map' : Π α, map (equiv.refl α) = @id (f α) . obviously)
+(map_map' : Π {α β γ} (k : α ≃ β) (h : β ≃ γ),
```

Comment:
Don't you want to call these `map_refl'` and `map_trans'`?

Reply:
Agreed. Fixed.